### ISSUE TITLE
UI showing more machines than what are available while spinning up downstream cluster

### ISSUE THREAD

**Issue by nwilliams22** [rancher/dashboard#6231]
**Setup**
- Rancher version: v2.6.3 (but have also seen it on v2.6.6-rc1)
- Installation option (Docker install/Helm Chart): Helm Chart
- RKE Version: v1.8.7

**Information about the Cluster**
- K8s Version: v1.21.13
- Cluster Type (Local/Downstream): 5 downstream node driver RKE clusters: 1 worker, 1 etcd, 1 cp

**Describe the bug**
Occasionally, while a downstream cluster is being built, the Rancher UI will show that there are more machines ready than there are machines for that node. (See screenshot at the bottom of the ticket). I've only seen it be +1, as in x number of machines and the UI shows "x+1 of x"

**To Reproduce**
This is an intermittent issue, but it seems to repro more often if you create a couple clusters in quick succession.

1. Spin up a Rancher instance (I've seen this repro on any version from 2.6.3 to 2.6.6-rc1)
2. Create several downstream clusters in quick succession (I did five EC2 clusters with the same setup and it was the fourth that showed this issue)
3. Go to the Cluster Management page
4. Watch as the clusters are being created

**Result**
The UI occasionally shows the count of available machines being more machines than there are in the cluster while that cluster is being made.

**Expected Result**
The UI never shows the count of available machines being more than the number of machines in the cluster while that cluster is being made.

**Screenshots**

![image](https://user-images.githubusercontent.com/28793924/175605566-88acb432-b81b-46d8-a843-795d1387e423.png)

**Comment by brandonsuse** [comment]
@nickwsuse Please reach out to @ronhorton if you need more context on this particular ticket. Thanks!

**Comment by nickwsuse** [comment]
`Verified on v2.6-head Commit ID: ed25172`

Since this one was intermittent I did the repro steps listed in the ticket by creating five downstream EC2 clusters in quick succession. I didn't see this issue occur again while those were being provisioned, so I made a few other clusters _not_ in quick succession and with various numbers of machines. They all showed the correct number of machines, so I think this one is good.
